### PR TITLE
[APM2-853] Add support for RMS wallets

### PR DESCRIPTION
**Pull request by FhanOmise**
#### 1. Objective

Add support RMS wallets payment methods
- Boost
- ShopeePay
- Maybank QRPAY
- DuitNow QR
- DuitNow Online Banking/Wallets

Update Touch 'n Go payment method 
-`TNG eWallet` for `provider: Alipay_plus`
-`Touch 'n Go` for default

Task: https://opn-ooo.atlassian.net/browse/APM2-853

#### 2. Description of change

Add RMS wallets payment methods config
Screen Shot 2565-07-21 at 18 10 18
Screen Shot 2565-07-21 at 18 10 03

Add RMS wallets payment methods on checkout page

Screen Shot 2565-07-21 at 17 58 51

#### 3. Quality assurance

- Checkout with added payment methods successfully
- touch_n_go payment method show title correctly for Alipay_plus and RMS provider.

**🔧 Environments:**

- **WooCommerce**: v6.7.0
- **WordPress**: v6.0.1
- **PHP version**: 7.3.3

#### 5. Priority of change

Normal.

### CODE REVIEW

File: includes/class-omise-capabilities.php [hunk 1]
```diff
@@ -61,6 +61,18 @@ public function getBackends( $currency = '' ) {
 		return $this->capabilities->getBackends( $params );
 	}
 
+	/**
+	 * Retrieves details of Touch n Go payment backends from capabilities.
+	 *
+	 * @return string
+	 */
+	public function getTouchNGoBackends() {
```
Comment (new4762) [inline includes/class-omise-capabilities.php:69]: do we really need this one as i say is will be 
```json
            "touch_n_go": {
                "type": "touch_n_go",
                "currencies": [
                    "jpy",
                    "sgd",
                    "myr",
                    "thb"
                ],
                "amount": {
                    "min": 3000,
                    "max": 1000000
                },
                "provider": "RMS"
            }
```

File: includes/backends/class-omise-backend-touch-n-go.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		$typeofFirstValue = gettype($first_value);
+		$isObjectOrString = 'object' === $typeofFirstValue || 'string' === $typeofFirstValue;
+
+        $provider = null;
+		if ($isObjectOrString && property_exists($first_value, 'provider')) {
+            $provider = $first_value->provider;
+		}
+
+        return $provider;
+	}
```
Comment (hpariwat) [inline includes/backends/class-omise-backend-touch-n-go.php:36]: I think the indentation for this file is a bit strange 🤔

File: includes/class-omise-capabilities.php (hunk 1 above)
Comment (FhanOmise) [reply includes/class-omise-capabilities.php:69]: I use it for switching the title here https://github.com/omise/omise-woocommerce/blob/b3e20886d48e3f77e411c6e477865dcaaff47a0c/includes/backends/class-omise-backend-touch-n-go.php#L21

File: includes/backends/class-omise-backend-touch-n-go.php (hunk 2 above)
Comment (FhanOmise) [reply includes/backends/class-omise-backend-touch-n-go.php:34]: Yea, it's strange. It look normal on VS Code 😅. Fixed [b3e2088](https://github.com/omise/omise-woocommerce/pull/287/commits/b3e20886d48e3f77e411c6e477865dcaaff47a0c)

**Review by hpariwat — APPROVED** [review]

File: templates/payment/form-duitnow-obw.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<fieldset id="omise-form-installment">
+	<div class="fpx-select-bank">
+		<label for="fpx-select-bank">Select Bank</label>			  
+		<select class="fpx-bank-logo default" id="duitnow-obw-select-bank" name="source[bank]">
+			<option value="" disabled selected>-- Select your option --</option>
+			<?php foreach ($viewData['duitnow_obw_banklist'] as $bank) : ?>
+				<option	
+					class="<?php echo $bank["code"];?>"
```
Comment (aashishgurung) [inline templates/payment/form-duitnow-obw.php:8]:
```suggestion
					class="<?= $bank['code'] ?>" 
```

File: templates/payment/form-duitnow-obw.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+		<label for="fpx-select-bank">Select Bank</label>			  
+		<select class="fpx-bank-logo default" id="duitnow-obw-select-bank" name="source[bank]">
+			<option value="" disabled selected>-- Select your option --</option>
+			<?php foreach ($viewData['duitnow_obw_banklist'] as $bank) : ?>
+				<option	
+					class="<?php echo $bank["code"];?>" 
+					value="<?php echo $bank["code"]; ?>"
+				>
+						<?php echo $bank["name"]; ?>
+				</option>
```
Comment (aashishgurung) [inline templates/payment/form-duitnow-obw.php:12]:
```suggestion
				<option	
					class="<?= $bank['code'] ?>" 
					value="<?= $bank['code'] ?>"
				>
						<?= $bank['name'] ?>
				</option>
```

File: includes/gateway/class-omise-payment-touch-n-go.php [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+		);
+	}
+		
+	public function GetMethodTitle() {
+		if ($this->backend->get_provider() === 'Alipay_plus') {
+			return 'TNG eWallet';
+		}
+		else {
+			return 'Touch \'n Go';
+		}
```
Comment (aashishgurung) [inline includes/gateway/class-omise-payment-touch-n-go.php:70]:
```suggestion
		if ($this->backend->get_provider() === 'Alipay_plus') {
			return 'TNG eWallet';
		}
	
		return 'Touch \'n Go';
```

Comment (FhanOmise) [reply includes/gateway/class-omise-payment-touch-n-go.php]: Fixed here [b7644cb](https://github.com/omise/omise-woocommerce/pull/287/commits/b7644cb10e4d354648ca7d837d989ab411d20f7f)

File: templates/payment/form-duitnow-obw.php (hunk 4 above)
Comment (FhanOmise) [reply templates/payment/form-duitnow-obw.php]: Fixed here [b7644cb](https://github.com/omise/omise-woocommerce/pull/287/commits/b7644cb10e4d354648ca7d837d989ab411d20f7f)

**Review by hpariwat — APPROVED** [review]

**Comment by tanawin-opn** [comment]
Tested and worked great krub. 👍

**Review by new4762 — APPROVED** [review]

**Pull request merged**